python/pyiceberg/expression/__init__.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from pyiceberg.expression.expressions import (
+    And,
+    Equal,
+    GreaterThan,
+    GreaterThanEqual,
+    In,
+    IsNaN,
+    IsNull,
+    LessThan,
+    LessThanEqual,
+    Not,
+    NotEqual,
+    NotIn,
+    NotNaN,
+    NotNull,
+    Or,
+)
+from pyiceberg.expression.literals import (
+    AboveMax,
+    BelowMin,
+    BinaryLiteral,
+    BooleanLiteral,
+    DateLiteral,
+    DecimalLiteral,
+    DoubleLiteral,
+    FixedLiteral,
+    FloatLiteral,
+    LongLiteral,
+    StringLiteral,
+    TimeLiteral,
+    TimestampLiteral,
+    UUIDLiteral,
+    literal,
+)
+
+__all__ = [
```

Comment:
I thought about this but the expression related stuff does have enough of a logical grouping with things like literals, evaluators, all of the visitors, etc. so it seems valuable to have all of this contained in a submodule for context. I'm ok with either way here though so if nobody else chimes in I'll update this. 👍

Reply:
I don't think literals really need to be exposed. Does that affect this decision?

Reply:
> I don't think literals really need to be exposed. Does that affect this decision?

If we replace the `expression` module with a single `expressions.py` file, should the contents of `literals.py` just be moved to the top of that file then? It would still be exposed if someone wanted to import it from `pyiceberg.expressions`.